samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemConsumer.java

```diff
@@ -330,6 +341,57 @@ public String getSystemName() {
     return systemName;
   }
 
+  @VisibleForTesting
+  static class KafkaStartpointVisitor implements StartpointVisitor {
```

Comment:
Could you please change the name to indicate it is responsible for startpoint registration? "StartpointVisitor" can be anything that needs to act on a `Startpoint`, so just using `KafkaStartpointVisitor` isn't specific enough.